js/parser.js

```diff
@@ -1,32 +1,244 @@
 /*
- * Copyright 2016 University of Southern California
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *    http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
+* Copyright 2016 University of Southern California
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
 
 var ERMrest = (function(module) {
 
     /**
-     * This is an internal function that parses a URI and constructs an
-     * internal representation of the URI.
-     * @memberof ERMrest
-     * @function _parse
-     * @private
-     */
-    module._parse = function(uri) {
-        // TODO parse a uri into some internal structure
+    * This is an internal function that parses a URI and constructs an
+    * internal representation of the URI.
+    * @memberof ERMrest
+    * @function _parse
+    * @param {String} uri - uri to be parsed
+    * @private
+    */
+    module._parse = function(uri, location) {
```

Comment:
and that will have an effect on the parsing logic below which was only parsing a fragment. so it will be a little different but i don't think it will be hard.

1. chaise parser needs to take the fragment and the location and basically glue them into an ermrest url, then
2. chaise will pass that url to the ermrestjs `resolve` function
3. then `resolve` passes it to `_parse`, and
4. `_parse` then does the real parsing

sorry for the confusion, this was maybe underspecified, but not a huge impact.